frontend/model/database.js

```diff
@@ -3,102 +3,114 @@
 import sbp from '@sbp/sbp'
 import { CURVE25519XSALSA20POLY1305, decrypt, encrypt, generateSalt, keyId, keygen, serializeKey } from '../../shared/domains/chelonia/crypto.js'
 
-const _instances = []
+const _instances: (() => Promise<*>)[] = []
 // Localforage-like API for IndexedDB
 const localforage = {
   async ready () {
-    await Promise.all(_instances).then(() => {})
+    await Promise.all(_instances.map((lazyInitDb) => lazyInitDb()))
   },
   createInstance ({ name, storeName }: { name: string, storeName: string }) {
     // Open the IndexedDB database
-    const db = new Promise((resolve, reject) => {
-      if (name.includes('-') || storeName.includes('-')) {
-        reject(new Error('Unsupported characters in name: -'))
-        return
-      }
-      const request = self.indexedDB.open(name + '--' + storeName)
+    const lazyInitDb = (() => {
```

Comment:
Could you write something to the effect of:

We lazy load the IndexedDB, because before we were loading it when this file was imported, but on iOS IndexedDB is not available in the service worker when the PWA is in the background <https://bugs.webkit.org/show_bug.cgi?id=283793>

So for that reason, and potentially other situations where IndexedDB might not be available, we lazy load it like this upon creation.